addOns/client/src/main/java/org/zaproxy/addon/client/ExtensionClientIntegration.java

```diff
@@ -863,4 +874,27 @@ public void sessionModeChanged(Mode mode) {
             }
         }
     }
+
+    public ClientMap getClientTree() {
+        return clientTree;
+    }
+
+    public void exportClientMap(String path) {
+        File file = new File(path);
+        Writer fileWriter = null;
+
+        try {
+            fileWriter = new FileWriter(file, false);
+        } catch (IOException e) {
```

Comment:
This should be part of the following try catch, if it failed it should not continue.